### PR TITLE
Add the title field to the sidebar

### PR DESCRIPTION
**Pull request by johannadevos**
## Summary

This PR can be summarized in the following changelog entry:

* [@yoast/schema-blocks] Removes the Job Title block in favor of an input field in the Job Posting sidebar.
* [@yoast/schema-blocks] Changes `'blocks'` to `'information'` in the sidebar.

## Relevant technical choices:

* I moved the QuestionMarkLink to a separate file and made it reusable by passing the URL as a prop.
* The SidebarHeader component also moved to a separate file, and is now used in `BlockDefinition` rather than `InnerBlocksSidebar`, so that it can be shown at the top of the sidebar.
* We no longer watch the post title, since we're no longer comparing the job title to it.

## Test instructions

This PR can be tested by following these steps:

* Test this PR together with https://github.com/Yoast/wordpress-seo-premium/pull/3290.
* Create a new post and add the Job Posting block.
* See that the Job Title block is gone.
* See that the sidebar now has an input field for the Job title.
* See that the analysis has a red bullet with the text "Job title has been left empty.".
* Fill in the other required blocks (description and location), and see that the analysis conclusion stays red as long as the job title hasn't been filled in.
* See that no Schema is generated on the front-end.
* Add white space to the job title field, and see that the analysis stays red.
* Add other content to the job title field (letters and/or numbers) and see that the warning about the job title disappears, and that the analysis turns green.
* See that Schema is generated on the front-end, and that the value for `title` corresponds to what you entered in the input field.
* See that the Job title label is accompanied by a clickable help icon, which is grey but turns blue when you hoover over it (compare the designs).

**Note**
* This PR contains a bug where you can only update the Job title field if the parent block is selected. However, because we are removing the parent block in https://yoast.atlassian.net/browse/P2-940 I'm not fixing this.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes
* [x] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Quality assurance

* [x] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

Fixes P2-866.

### CODE REVIEW

File: packages/schema-blocks/css/schema-blocks.css [hunk 1]
```diff
@@ -63,22 +63,31 @@
 	margin-right: 24px;
 }
 
-.yoast-block-sidebar-title {
+.yoast-block-sidebar-title, .yoast-block-sidebar-input label {
```
Comment (increddibelly) [inline packages/schema-blocks/css/schema-blocks.css:66]:
```suggestion
.yoast-block-sidebar-input label {
```

Comment (increddibelly) [reply packages/schema-blocks/css/schema-blocks.css:66]: die kan nu weg denk ik ?

File: packages/schema-blocks/src/functions/presenters/StringWithClickableQuestionMarkPresenter.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { createElement } from "@wordpress/element";
+import { QuestionMarkLink } from "./QuestionMarkLinkPresenter";
+
+interface StringWithClickableQuestionMark {
```
Comment (increddibelly) [inline packages/schema-blocks/src/functions/presenters/StringWithClickableQuestionMarkPresenter.tsx:4]:
```suggestion
interface LabelWithHelpLinkProps {
```

File: packages/schema-blocks/src/functions/presenters/StringWithClickableQuestionMarkPresenter.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+/**
+ * Renders a string followed by a clickable question mark icon.
+ *
+ * @param props The properties.
+ *
+ * @returns A ReactElement containing the string followed by a clickable question mark icon.
+ */
+export default function StringWithClickableQuestionMark( props: StringWithClickableQuestionMark ): JSX.Element {
```
Comment (increddibelly) [inline packages/schema-blocks/src/functions/presenters/StringWithClickableQuestionMarkPresenter.tsx:16]:
```suggestion
export default function LabelWithHelpLink ( props: LabelWithHelpLinkProps ): JSX.Element {
```

File: packages/schema-blocks/src/instructions/blocks/InheritSidebar.tsx [hunk 4]
```diff
@@ -8,7 +8,7 @@ import { getBlockDefinition } from "../../core/blocks/BlockDefinitionRepository"
 import { InstructionOptions } from "../../core/Instruction";
 
 /**
- * Sidebar input instruction.
+ * InheritSidebar instruction.
```
Comment (increddibelly) [inline packages/schema-blocks/src/instructions/blocks/InheritSidebar.tsx:11]: 😄

File: packages/schema-blocks/src/functions/presenters/StringWithClickableQuestionMarkPresenter.tsx (hunk 3 above)
Comment (increddibelly) [reply packages/schema-blocks/src/functions/presenters/StringWithClickableQuestionMarkPresenter.tsx:16]: het tonen van een Label is het doel van deze code. Dat het een String is, is een implementatie detail, wat de buitenwereld niets aangaat.

File: packages/schema-blocks/src/instructions/blocks/SidebarInput.ts [hunk 5]
```diff
@@ -21,21 +22,32 @@ class SidebarInput extends SidebarBase {
...
 		}
+
 		if ( this.options.type ) {
-			attributes.type = this.options.type;
+			textControlProps.type = this.options.type;
+		}
+
+		// If a help link was passed in the template, add the question mark icon with the link to the label.
+		if ( this.options.helpLink ) {
+			textControlProps.label = StringWithClickableQuestionMark( { text: textControlProps.label as string, URL: this.options.helpLink } );
```
Comment (increddibelly) [inline packages/schema-blocks/src/instructions/blocks/SidebarInput.ts:47]: label 😇

File: packages/schema-blocks/src/functions/presenters/QuestionMarkLinkPresenter.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+/**
+ * Renders a ReactElement containing a clickable question mark icon from HeroIcons.
+ *
+ * @param props The properties.
+ *
+ * @returns A ReactElement containing the question mark icon with a link.
+ */
+export function QuestionMarkLink( props: QuestionMarkLinkProps ): JSX.Element {
```
Comment (increddibelly) [inline packages/schema-blocks/src/functions/presenters/QuestionMarkLinkPresenter.tsx:14]:
```suggestion
export function HelpLink( props: QuestionMarkLinkProps ): JSX.Element {
```

**Review by increddibelly — CHANGES_REQUESTED** [review]

File: packages/schema-blocks/css/schema-blocks.css (hunk 1 above)
Comment (johannadevos) [reply packages/schema-blocks/css/schema-blocks.css:66]: Die styling blijft op zich relevant, ook als we de velden in de metabox gaan tonen (het `Job title` label moet dan nog steeds styling hebben). We moeten dan wel op een gegeven moment die CSS-classes gaan hernoemen naar `metabox`, oid, maar dat zou ik niet nu doen. Dat zijn er namelijk veel meer dan alleen deze.

File: packages/schema-blocks/src/functions/presenters/StringWithClickableQuestionMarkPresenter.tsx (hunk 2 above)
Comment (johannadevos) [reply packages/schema-blocks/src/functions/presenters/StringWithClickableQuestionMarkPresenter.tsx:4]: Done 👍

File: packages/schema-blocks/src/instructions/blocks/SidebarInput.ts (hunk 5 above)
Comment (johannadevos) [reply packages/schema-blocks/src/instructions/blocks/SidebarInput.ts:47]: Done

File: packages/schema-blocks/src/functions/presenters/QuestionMarkLinkPresenter.tsx (hunk 6 above)
Comment (johannadevos) [reply packages/schema-blocks/src/functions/presenters/QuestionMarkLinkPresenter.tsx:14]: Done

File: packages/schema-blocks/src/instructions/blocks/Title.tsx [hunk 7]
```diff
@@ -36,49 +26,39 @@ class Title extends Heading {
...
 	 */
 	validate( blockInstance: BlockInstance ): BlockValidationResult {
-		const blockTitle: string = blockInstance.attributes[ this.options.name ];
-		const postTitle: string = select( "core/editor" ).getEditedPostAttribute( "title" );
+		const title: string = blockInstance.attributes[ this.options.name ];
 
-		if ( ! this.isCompleted( blockInstance ) ) {
-			const presence = this.options.required === true ? BlockPresence.Required : BlockPresence.Recommended;
-			return BlockValidationResult.MissingAttribute( blockInstance, this.constructor.name, presence );
+		if ( title && ( title as string ).trim() ) {
```
[the remark this answers is not in the document]
Comment (johannadevos) [reply packages/schema-blocks/src/instructions/blocks/Title.tsx]: Implemented by @hansjovis

**Comment by johannadevos** [comment]
The Core squad has decided to merge this PR and its counterpart Yoast/wordpress-seo-premium#3290 without an acceptance test at this point, because `trunk` was already merged into the Premium PR, which makes it hard to test.

I have seen it working on my machine ✅

**Pull request merged**